google/auth/transport/requests.py

```diff
@@ -150,11 +151,25 @@ def __init__(self, credentials,
         self.credentials = credentials
         self._refresh_status_codes = refresh_status_codes
         self._max_refresh_attempts = max_refresh_attempts
+
+        auth_request_session = requests.Session()
+
+        # This adapter catches exceptions at errors on network layers
+        # lower than HTTP and retrys credentials.refresh().
+        # These exceptions include ConnectionError, ConnectTimeout.
+        # We shouldn't let them bubble up to client codes of this library
+        # because the clients cannot distinguish them with these from
+        # their own HTTP request calls and determine whether they can
+        # safely retry the calls which might cause non-idempotent
+        # HTTP requests as well as the retriable credential refresh.
+        retry_adapter = requests.adapters.HTTPAdapter(max_retries=3)
```

Comment:
Passing `urllib3.Retry(total=3, method_whitelist=False)` to the`max_retries` parameter makes the adapter also retry on the `ReadTimeout` exception, which reduce the `except:` block from the `request()` method.

Which do you think better?